package mpsqual;

message GPIORequest {
	enum RequestTypeT {
		DISCONNECT = 0;
		CONNECT	= 1;
		REPORT = 2;
	}
	required RequestTypeT requestType = 1 [default = DISCONNECT];
	required string gpIn  = 2 [default = ""];
	optional string gpOut = 3 [default = ""];
}

message GPIOResponse {
	enum ConStateT {
		DISCONNECTED = 0;
		CONNECTED = 1;
	}
	
	message GpioStatus {
		required ConStateT conState = 1;
		required int32 matchCount = 2;
		required int32 mismatchCount = 3;
		required string gpIn = 4;
		optional string gpOut = 5;
	}
	repeated GpioStatus status = 1;
}